akka-stream/src/main/scala/akka/stream/impl/fusing/GraphInterpreter.scala

```diff
@@ -302,6 +310,22 @@ private[stream] final class GraphInterpreter(
   }
 
   /**
+   * Dynamic handler changes are communicated from a GraphStageLogic by this method.
+   */
+  def setHandler(connection: Int, handler: InHandler): Unit = {
```

Comment:
oh, I didn't expect these. But of course you are right, the interpreter has everything flattened into a connection id indexed array.